Libraries/LibWeb/ContentSecurityPolicy/Directives/RequireTrustedTypesFor.cpp

```diff
@@ -105,4 +106,76 @@ bool does_sink_require_trusted_types(JS::Object& global, String sink_group, Incl
     // 2. Return false.
     return false;
 }
+
+// https://w3c.github.io/trusted-types/dist/spec/#should-block-sink-type-mismatch
+Directive::Result should_sink_type_mismatch_violation_be_blocked_by_content_security_policy(JS::Object& global, TrustedTypes::InjectionSink sink, String sink_group, Utf16String source)
+{
+    auto& realm = *global.vm().current_realm();
+
+    // 1. Let result be "Allowed".
+    auto result = Directive::Result::Allowed;
+
+    // 2. Let sample be source.
+    auto sample = source.substring_view(0);
+
+    // 3. If sink is "Function", then:
+    if (sink == TrustedTypes::InjectionSink::Function) {
+        // 1. If sample starts with "function anonymous", strip that from sample.
+        if (sample.starts_with("function anonymous"sv)) {
+            sample = sample.substring_view("function anonymous"sv.length());
```

Comment:
This pattern is a little spooky. `sample` is utf16, but I believe the `"function anonymous"sv` will be a plain `StringView`, which just counts bytes. (We seem to have definitions of `operator""sv` for both `StringView` and `Utf16StringView` somehow?) So you're turning a byte length into a code-point offset, which aren't the same thing.

cc @trflynn89, I haven't been following the Utf16 changes so maybe I'm wrong on this?

Reply:
I will be honest this also confused me. I didn't knew it was legal for the same operator to be defined twice. I feel like we are doing something illegal here.